### PR TITLE
[web] Update view with sortedUpdate

### PR DESCRIPTION
**Pull request by gzzhanghao**
Fix https://github.com/mitmproxy/mitmweb/issues/36

### CODE REVIEW

**Comment by mhils** [comment]
Awesome. Can we add a test for this and not add the compiled assets in the PR?

File: web/src/js/__tests__/ducks/tutils.js [hunk 1]
```diff
@@ -10,3 +10,9 @@ export function createStore(parts) {
         applyMiddleware(...[thunk])
     )
 }
+
+describe('tutils', () => {
```
Comment (mhils) [inline web/src/js/__tests__/ducks/tutils.js]: What's the point of this? :smiley:

File: web/src/js/ducks/utils/view.js [hunk 2]
```diff
@@ -54,21 +54,30 @@ export default function reduce(state = defaultState, action) {
...
-                ...nextState,
-                ...sortedInsert(nextState, action.item, action.sort)
+            if (hasNewItem && hasOldItem) {
+                return {
+                    ...state,
+                    ...sortedUpdate(state, action.item, action.sort),
+                }
             }
-
+            break;
```
Comment (mhils) [inline web/src/js/ducks/utils/view.js]: We should never hit this, right?

**Comment by mhils** [comment]
Thanks for the changes, but why is ever testfile called *Spec now? How about we make a negative RegEx just excluding the tutils?

**Comment by gzzhanghao** [comment]
I think it's better to have a general method to distinguish test files and utilities, we may have `__tests__/utils/*.js` sometimes later. :)

**Pull request merged**